File: stage3.py
import api
import json

def runstage3(token):
	# get data
	jsonData = api.getData('prefix', {'token':token})

	# extract data
	array = jsonData['array']
	prefix = jsonData['prefix']

	# For a problem like this I like ruby's nice array methods like 'reject' and collect
	# so I will 'simulate them'
	matches = [match for match in array if not match.startswith(prefix)]

	# validate result
	return api.verifyResult('validateprefix', {'token':token, 'array':matches})

File: ruby/stage3.rb
require 'http'
require './api'

module Stage3
	def self.runstage3(token)
		# get data
		data = API.communicate('prefix', {'token'=> token})

		# extract data
		words = data['array']
		prefix = data['prefix']

		# I LOVE the functional paradigms that ruby has borrowed
		matches = words.reject { |word| word.start_with? prefix }

		# validate result
		API.communicate('validateprefix', {'token'=> token, 'array'=> matches})
	end
end